components/bitcoind/src/utils/redis_notifier.rs

```diff
@@ -0,0 +1,362 @@
+use std::sync::{Mutex, OnceLock};
+
+use config::{BitcoindConfig, RedisConfig};
+use redis::{AsyncCommands, Client};
+use serde::Serialize;
+
+use crate::types::BlockIdentifier;
+
+#[derive(Serialize)]
+struct IndexProgressPayload<'a> {
+    chain: &'a str,
+    network: &'a str,
+    indexer: &'a str,
+    apply_blocks: Vec<BlockIndexRef>,
+    rollback_blocks: Vec<BlockIndexRef>,
+}
+
+#[derive(Serialize)]
+struct Message<'a> {
+    id: String,
+    payload: IndexProgressPayload<'a>,
+}
+
+#[derive(Serialize)]
+struct BlockIndexRef {
+    hash: String,
+    index: u64,
+}
+
+pub struct RedisNotifier {
+    pub queue: String,
+    pub client: Client,
+}
+
+impl RedisNotifier {
+    pub fn new(redis: &RedisConfig) -> Result<Self, String> {
+        let client = Client::open(redis.url.as_str())
+            .map_err(|e| format!("unable to connect to redis: {e}"))?;
```

Comment:
How does this client handle reconnections? Also, we need to add support for High Availability Redis (Sentinels or Cluster) in these connection options